### PR TITLE
PAT-328 Only consider approved risk_banding_assessments when determin…

### PR DESCRIPTION
**Pull request by yasinmustafa**
…ing the highest previous band

### CODE REVIEW

File: server/data/pathfinderClient.js [hunk 1]
```diff
@@ -94,7 +95,7 @@ const createRiskBandingAssessment = async (nominalId, status, username, form, ba
   return result.rows[0].id
 }
 
-const getRiskBandingAssessmentByNominalId = async nominalId => {
+const getRiskBandingAssessmentsByNominalId = async nominalId => {
```
Comment (timharrison-moj) [inline server/data/pathfinderClient.js]: The return value is still the first row[0].id - is it still meant to be plural i.e. 'assessments'?

File: server/routes/review.js [hunk 2]
```diff
@@ -108,22 +108,26 @@ module.exports = ({ offenderService, pathfinderService }) => router => {
       const context = { username: res.locals.user.username, token: res.locals.user.token }
       const nominalData = await pathfinderService.getNominalById(nominalId)
       const offenderBasic = await offenderService.getOffenderBasicByBookingId(context, nominalData.bookingId)
-      const riskBandingAssessment = await pathfinderService.getRiskBandingAssessmentByNominalId(nominalData.id)
-      res.render('pages/review/bandingReviews', { offenderBasic, nominalData, riskBandingAssessment })
+      const riskBandingAssessments = await pathfinderService.getRiskBandingAssessmentsByNominalId(nominalData.id)
+      res.render('pages/review/bandingReviews', {
```
Comment (timharrison-moj) [inline server/routes/review.js]: Same as above - still a singular row id?

File: server/routes/review.js [hunk 3]
```diff
@@ -108,22 +108,26 @@ module.exports = ({ offenderService, pathfinderService }) => router => {
...
   )
 
   router.post(
     '/bandingReviews/:nominalId',
     asyncMiddleware(async (req, res) => {
       const { nominalId } = req.params
-      const riskBandingAssessment = await pathfinderService.getRiskBandingAssessmentByNominalId(nominalId)
-      if (riskBandingAssessment.length === 0 || riskBandingAssessment[0].status !== NEW) {
+      const riskBandingAssessments = await pathfinderService.getRiskBandingAssessmentsByNominalId(nominalId)
+      if (riskBandingAssessments.length === 0 || riskBandingAssessments[0].status !== NEW) {
```
Comment (timharrison-moj) [inline server/routes/review.js]: Is this returning what you are expecting here? Just seeing the id coming back..

File: server/data/pathfinderClient.js (hunk 1 above)
Comment (yasinmustafa) [reply server/data/pathfinderClient.js]: which line number are you referring to?  it returns result.rows (line 117)

If you mean line 94 , that';s actually a different method (unchanged)

File: server/routes/review.js (hunk 2 above)
Comment (yasinmustafa) [reply server/routes/review.js]: it does return many (result.rows)

File: server/routes/review.js (hunk 3 above)
Comment (yasinmustafa) [reply server/routes/review.js]: it does return many (result.rows)

File: server/data/pathfinderClient.js (hunk 1 above)
Comment (timharrison-moj) [reply server/data/pathfinderClient.js]: And its me!  I skimmed it at the summary level and there's 200 lines between them! Sorry, looks ok when viewed in full.

**Pull request merged**